docs/src/architecture/domain-model.md

```diff
@@ -0,0 +1,184 @@
+
+
+## Plugins
+
+**thin-edge** realizes **Device Management** based on **plugins**.
+  * a **plugin** can be an (external) executable (e.g. as the `c8y_configuration_plugin` for _configuration management_)
+    or a thin-edge built-in software component (e.g. as for _software management_)
+  * to realize **Device Management** functionality, a **plugin** accesses _resources_ and _services_ of the device, as e.g.
+      * _software management_ uses the device's _package manager_
+      * _configuration management_ reads/writes device's _configuration files_
+  * a **plugin** can facilitate **Device Management** functionality for the **main-device** and **external child-device**, but runs usually only on the **main-device**
+    * _resources_ and _services_ (e.g. file system, package manager, ...) of the **main-device** can be directly accessed by the **plugin**, as it runs on the **main-device**
+    * all _resources_ and _services_ of an **external child-device** are only remotely accessible for the **main-device**
+      * the **plugin** running on the **main-device** needs another software component that establish the remote access for those _resources_ and _services_
+      * that software component is referred as **child-device agent**
+  * a **plugin** can also facilitate **Device Management** functionality for containers running on the **main-device**
+    * similar to **external child-devices**, all _resources_ and _services_ of a container are accessed by a **plugin** via a **child-device agent** that runs inside the corresponding container 
+
+The figure below illustrates the concept of **plugins** and **child-devices agents**.
+
+![Plugin Concept](images/plugin-concept.svg)
+
+### Child-Device Agent
+  * a **child-device agent** is the counterpart of a **plugin**, that takes the responsibility to access to the **external child-device's** _resources_
+  * a **child-device agent** can serve one or more **plugins**
+  * a **child-device agent** can be installed and executed on the **external child-device**, or on the **main-device**
+    * if it runs in the **external child-device** it can access the _resources_ directly
+    * if it runs on the **main-device** it can use any (low-level) interfaces the **external child-device** provides to access those resources
+      * One main reason to install the **child-device agent** on the **main-device** is, when the **external child-device** cannot or shall not be altered.
+
+### Plugin-Contract
+
+A **plugin** defines and implements a specific **contract** for all interactions with a **child-device agent**
+  * part of the **contract** could be e.g.:
+      * the **child-device agent** must listen and react to certain requests of the **plugin**, e.g. on MQTT
+      * the **child-device agent** must provide/consume files to/from the **plugin** on purpose, e.g. via HTTP
+      * ...and more...
+  * a **plugin's** **contract** can be flagged with a unique name (e.g. `tedge_config`)
```

Comment:
How about calling this unique name a `plugin-capability` instead of `plugin-contract`. A `plugin-capability`.

Reply:
True, "plugin-contract name" doesn't sound perfect. 

> How about calling this unique name a plugin-capability instead of plugin-contract.

I think term "plugin-capability" is also not the right thing. Cause the "contract" is not a capabiity, but instead the whole plugin might be more a "capability".

However to address you point I introduced now the term "plugin-identifier" as name for the whole plugin, and removed the "plugin-contract name". See https://github.com/thin-edge/thin-edge.io/pull/1554/commits/98bd39eb0c536930363ee7e271da21598e0b1b98

I feel that improves that section a lot.